Pinta.Core/Classes/Document.cs

```diff
@@ -153,7 +153,12 @@ public UserLayer CurrentUserLayer
 		/// Directory and file name, like "C:\MyPictures\dog.jpg".
 		/// </summary>
 		public string PathAndFileName {
-			get { return System.IO.Path.Combine (Pathname, Filename); }
+			get {
+				if (!string.IsNullOrEmpty(Pathname) && !string.IsNullOrEmpty(Filename))
+					return System.IO.Path.Combine (Pathname, Filename);
```

Comment:
When I created 'FullContentName' for the DocumentViewContent, the application immediately crashed on startup because it was opening an 'Unsaved Image' and was accessing the 'PathAndFileName' property, which had no Pathname and System.IO.Path.Combine required a non-null value. I resolved that problem by just adding empty string checks to both.